corehq/apps/geospatial/tasks.py

```diff
@@ -14,3 +29,47 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@task(queue='geospatial_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, ES_INDEX_TASK_HELPER_BASE_KEY)
+    geo_case_prop = get_geo_case_property(domain)
+    query = case_query_for_missing_geopoint_val(domain, geo_case_prop)
+    doc_count = query.count()
+    if doc_count > MAX_GEOSPATIAL_INDEX_DOC_LIMIT:
+        celery_task_tracker.mark_as_error(error_slug='TOO_MANY_CASES')
+        return
+
+    celery_task_tracker.mark_requested()
+    batch_count = math.ceil(doc_count / DEFAULT_QUERY_LIMIT)
+    try:
+        for i in range(batch_count):
+            docs_left = doc_count - (DEFAULT_QUERY_LIMIT * i)
+            limit = min(DEFAULT_QUERY_LIMIT, docs_left)
+            process_batch(
+                domain,
+                geo_case_prop,
+                case_type=None,
+                query_limit=limit,
```

Comment:
is it needed for us to calculate the remaining docs and pass that here?
I believe if you just pass DEFAULT_QUERY_LIMIT even in the last iteration, it would work?

nit: I'd suggest renaming that variable to "remaining" instead of limit since that is what it is.

Reply:
Testing this out locally it works fine with specifying a size higher than what is remaining. Given this, I'll simplify this as we don't need to calculate the remaining docs.

Reply:
Addressed in [93f259a](https://github.com/dimagi/commcare-hq/pull/35126/commits/93f259acbe27cfb4c92f1fe5bc92f29bb01aa00f)